src/main/java/com/smartbear/tcpbench/rtptorrent/RtpTorrentQuery.java

```diff
@@ -86,6 +99,38 @@ public static Query create(File rtpTorrentProjectDir) throws Exception {
         return patches.stringColumn("name").asSet();
     }
 
+    @Override
+    public Changes getChanges(String oldSha, String newSha, String regexp) {
+        ObjectId oldCommitId = ObjectId.fromString(oldSha);
+        ObjectId newCommitId = ObjectId.fromString(newSha);
+        DiffFormatter diffFormatter = new DiffFormatter(DisabledOutputStream.INSTANCE);
+        diffFormatter.setRepository(git.getRepository());
+        diffFormatter.setDiffComparator(RawTextComparator.DEFAULT);
+        int linesAdded = 0;
+        int linesDeleted = 0;
+        int timeDiff = 0;
+        int changedFiles = 0;
+        try (RevWalk revWalk = new RevWalk(git.getRepository())) {
+            RevCommit oldCommit = revWalk.parseCommit(git.getRepository().resolve(oldSha));
+            RevCommit newCommit = revWalk.parseCommit(git.getRepository().resolve(newSha));
+            timeDiff = newCommit.getCommitTime() - oldCommit.getCommitTime();
+            List<DiffEntry> diffEntries;
+            diffEntries = diffFormatter.scan(oldCommitId, newCommitId).stream().filter(diffEntry -> diffEntry.getNewPath().matches(regexp) || diffEntry.getOldPath().matches(regexp)).collect(toList());
+            changedFiles = diffEntries.size();
+            for (DiffEntry diffEntry : diffEntries) {
+                for (Edit edit : diffFormatter.toFileHeader(diffEntry).toEditList()) {
+                    if (edit.getType() != Edit.Type.REPLACE) {
+                        linesDeleted += edit.getEndA() - edit.getBeginA();
+                        linesAdded += edit.getEndB() - edit.getBeginB();
+                    }
+                }
+            }
+            return new Changes(changedFiles, linesAdded, linesDeleted, timeDiff);
+        } catch (IOException e) {
+            return new Changes();
```

Comment:
If an IOException happens I think we want to know about it, rather than silently pretending everything is OK. I'd recommend throwing it.

Reply:
It happens sometimes a commit is not found for any reasons, I ignore this exception to avoid stopping execution for this rare case. But we can throw it and see if the execution is stopped to often, we can also skip cycles concerned by this rare issue.